File: clang/test/SemaCXX/delete.cpp
// RUN: %clang_cc1 -fsyntax-only -verify %s
// RUN: cp %s %t
// RUN: %clang_cc1 -fixit -x c++ %t
// RUN: %clang_cc1 -E -o - %t | FileCheck %s

void f(int a[10][20]) {
  // CHECK: delete[] a;
  delete a; // expected-warning {{'delete' applied to a pointer-to-array type}}
}

File: llgo/test/execution/maps/delete.go
// RUN: llgo -o %t %s
// RUN: %t 2>&1 | FileCheck %s

// CHECK: 1
// CHECK-NEXT: 1 1
// CHECK-NEXT: 0 0

package main

func main() {
	m := make(map[int]int)
	delete(m, 0) // no-op
	m[0] = 1
	println(len(m))
	delete(m, 1) // no-op
	println(len(m), m[0])
	delete(m, 0) // delete element in map
	println(len(m), m[0])
}
